packages/core/src/common/performance/measurement.ts

```diff
@@ -101,4 +101,30 @@ export interface MeasurementOptions {
      * @see {@link thresholdLogLevel}
      */
     thresholdMillis?: number;
+
+    /**
+     * Flag to indicate whether the stopwatch should cache measurement results for later retrieval.
+     * For example the cache can be used to retrieve measurements which were taken during startup before a listener had a chance to register.
+     */
+    cacheResults?: boolean
```

Comment:
In IT, the word "cache" usually refers to a short-term, incomplete set of items, often size-limited,  where some items may be replaced according to some algorithm. I suspect this is not what's happening here. If we want to store a full trace of results, we should call this field `storeResults` and rename related variables accordingly.